### ISSUE TITLE
[Feature] Search bar should prioritise exact file name matches over fuzzy matches

### ISSUE THREAD

**Issue by thomasaarholt** [dbt-labs/dbt-docs#217]
### Describe the bug
Searching for a model e.g. `f_my_model` often returns results where the model in question is surprisingly far down the list.
We often have models with similar names with different underscores, for instance `a_users` and `b_users` or `f_orders_current` and `h_orders_old` (I have had to make up some names here).

### Steps To Reproduce
Open the generated dbt documentation, and search for a model name in the top bar. Type the exact model name (minus extension), and observe that the model is not always the first hit.  

### Expected behavior
Exact model name should be listed at the top.

### Screenshots and log output
I can't provide any screenshots due to it being a private dbt project, but just to provide explicit context, this is the search field I mean:
![image](https://user-images.githubusercontent.com/2721423/136405879-27a3112e-ad64-4fea-ba6b-4223d8d3ea92.png)

**The output of `dbt --version`:**
```
installed version: 0.20.2
   latest version: 0.21.0

Your version of dbt is out of date! You can find instructions for upgrading here:
https://docs.getdbt.com/docs/installation

Plugins:
  - bigquery: 0.20.2
  - snowflake: 0.20.2
  - redshift: 0.20.2
  - postgres: 0.20.2```

### Additional context
Add any other context about the problem here.

**Comment by dbeatty10** [comment]
Agreed that this should not be labeled as "completed" -- thanks for flagging this @rightx2 👍

**Comment by dbeatty10** [comment]
Re-categorizing this as a feature request and labeling as "help wanted".

**Comment by sercancicek** [comment]
Created a [PR](https://github.com/dbt-labs/dbt-docs/pull/503) for this issue
